### ISSUE TITLE
Current working directory complete in addition to history complete

### ISSUE THREAD

**Issue by ehawman** [nushell/nushell#5799]
### Related problem

I have read the Reedline documentation but either the feature isn't there or how to accomplish it isn't clicking in my head.

In Fish, the default configuration suggests the contents of the current working directory in addition to history.

Is this an existing option and I'm just wooshing on it hard? If it is, can it be made default?

### Describe the solution you'd like

Use both history and the CWD to make autocomplete suggestions

### Additional context and details

![image](https://user-images.githubusercontent.com/52979227/173901896-22b34866-6fe9-460e-bdae-9186eea834fa.png)

https://asciinema.org/a/8g9NWwULdsus6xyIjF5ztmMMs

**Comment by fdncred** [comment]
i don't think we have this feature of searching the $env.PWD for completion, although it seems like a good idea to me.

**Comment by sholderbach** [comment]
We should have file completions. I see when hitting tab after e.g. `open ` it tries to matches commands instead of files in the cwd..

**Comment by sholderbach** [comment]
Ah sorry I think I misunderstood. So this is just an inline hint like the history autosuggestions. Sorry for the confusion. That would be nice. Would require to implement a `Reedline::Hinter`.

**Comment by danjenson** [comment]
I feel like this is a regression with 0.64 -- I thought this was working for 0.63 and came to report that it had stopped working with the recent upgrade.

**Comment by sholderbach** [comment]
Let me move my comment into a separate issue because I consider that to be a bug, but I think the fish feature is a nice enhancement somebody can pick up to implement if they find the time and fun

**Comment by Kangaxx-0** [comment]
> I feel like this is a regression with 0.64 -- I thought this was working for 0.63 and came to report that it had stopped working with the recent upgrade.

I just tested 0.63, it did not work. I have to type something like `ls D<tab>` to see the DIRS

**Comment by UdeshyaDhungana** [comment]
I think this can be done using `hinter.closure` now.

Here's a sample:
```nu
def get-current-word [line: string, pos: int] {
    $line | str substring ..<$pos | split row -r '\s+' | last
}

$env.config.hinter.closure = {|ctx|
    if ($ctx.line | is-empty) { return null }

    let word = get-current-word $ctx.line $ctx.pos

    let hist_matches = (
        history
        | get command
        | where ($it | str starts-with $ctx.line) and (($it | str length) > ($ctx.line | str length))
    )

    if not ($hist_matches | is-empty) {
        return ($hist_matches | last | str substring ($ctx.line | str length)..)
    }

    if ($word | is-empty) { return null }

    let matches = (
        ls $ctx.cwd
        | get name
        | path basename
        | where ($it | str starts-with $word)
        | sort
    )
    if ($matches | is-empty) { return null }

    let suffix = ($matches | first | str substring ($word | str length)..)
    if ($suffix | is-empty) { null } else { $suffix }
}
```

Are we trying to make this a default or leave it as it is since it can be configured by the user?
